src/inventory_ui.cpp

```diff
@@ -781,19 +967,28 @@ int inventory_selector::execute_pick( const std::string &title, const int positi
 
         if( itemstack != nullptr ) {
             return itemstack->item_pos;
-        } else if ( handle_movement( action ) ) {
-            continue;
         } else if ( action == "CONFIRM" || action == "RIGHT" ) {
-            return get_selected_item_position();
+            const auto selection( get_active_column().get_all_selected() );
+
+            if( selection.empty() ) {
+                return INT_MIN;
+            } else if( selection.size() == 1 ) {
+                return selection.front()->item_pos;
+            } else {
+                const std::string msg( _( "Please pick only one item." ) );
+                popup( msg , PF_GET_KEY );
```

Comment:
This shouldn't really happen.
When the selector is in single-select mode, it should not look like it is in multi-select mode and should not allow selecting multiple items. Making it default to the first one is fine, as long as it displays only the first one as selected.

Also, "please" in user commands is a bit too polite. It sounds like we're giving the user a choice, which isn't the case. "Only one item can be selected" or something like that.